artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPMessage.java

```diff
@@ -642,9 +657,17 @@ protected synchronized void scanMessageData() {
             } else if (DeliveryAnnotations.class.equals(constructor.getTypeClass())) {
                // Don't decode these as they are not used by the broker at all and are
                // discarded on send, mark for lazy decode if ever needed.
-               constructor.skipValue();
-               deliveryAnnotationsPosition = constructorPos;
-               encodedDeliveryAnnotationsSize = data.position() - constructorPos;
+               try {
+                  deliveryAnnotationsPosition = constructorPos;
+                  this.deliveryAnnotations = (DeliveryAnnotations) constructor.readValue();
+                  encodedDeliveryAnnotationsSize = data.position() - constructorPos;
+               } catch (Exception e) {
+                  // mal formed delivery annotation, we will just skip it then
+                  logger.debug(e.getMessage(), e);
+                  data.position(constructorPos);
+                  constructor.skipValue();
+                  encodedDeliveryAnnotationsSize = data.position() - constructorPos;
+               }
```

Comment:
@gemmellr I will fix the comment...


The skip was actually added because one of the unit tests is adding and forcing an invalid delivery annotation...

So, I'm just keeping this to not change semantics.

Reply:
@gemmellr I am using Delivery Annotations over the mirror to send the ID a message will receive over the ACK part.

Reply:
I dont think thats a good reason to skip this. It may just mean the test needs changed, as the brokers behaviour has in fact substantially changed here. Which test? 

I know you are using the delivery annotations now, which is part of my point. They are being used for something, and if something somehow makes it invalid (e.g a concurrency issue, of which there have been plenty), this will then skip them silently, which could mean it doesnt do something that it should have and which was intended for it to do.

Its a decode failure, it should fail the way it would any other decode failure.

Reply:
@gemmellr AMQPMessageTest::testPartialDecodeIgnoresDeliveryAnnotationsByDefault is testing explicitly for this scenario

https://github.com/apache/activemq-artemis/blob/69e58322f1b99c742a2a9761d92d61e955e535a5/artemis-protocols/artemis-amqp-protocol/src/test/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPMessageTest.java#L1587-L1625

Reply:
This test was originally checking that since the broker didn't ever use any delivery annotations and simply trimmed them that it didn't bother to try and decode and validate them  Since you are actually using them in this new feature it probably means you can now remove this test and react accordingly to delivery annotations that have encoding issues vs just trimming them out.

Reply:
Yep, now thats been established, removing the 'not used, no impact' test makes sense, replacing it with an 'they are used and so there is a reaction' one.